init/cmdline.go

```diff
@@ -127,23 +127,29 @@ func parseParams(params string) error {
 				return fmt.Errorf("invalid UUID %s %v", parts[0], err)
 			}
 
-			dev := luksMapping{
-				ref:  &deviceRef{refFsUUID, uuid},
-				name: parts[1],
-			}
-			luksMappings = append(luksMappings, dev)
+			var o = findOrCreateLuksMapping(uuid)
+			o.name = parts[1]
 		case "rd.luks.uuid":
 			stripped := stripQuotes(value)
-			u, err := parseUUID(stripped)
+			uuid, err := parseUUID(stripped)
 			if err != nil {
 				return fmt.Errorf("invalid UUID %s in rd.luks.uuid boot param: %v", value, err)
 			}
 
-			dev := luksMapping{
-				ref:  &deviceRef{refFsUUID, u},
-				name: "luks-" + stripped,
+			findOrCreateLuksMapping(uuid)
```

Comment:
It looks like setting name to `name: "luks-" + stripped,` is lost here.

Reply:
findOrCreate defaults to name = luks-UUID when it creates